core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -133,44 +134,49 @@ protected WritePrimaryResult shardOperationOnPrimary(BulkShardRequest request, I
             // execute item request
             final Engine.Result operationResult;
             final DocWriteResponse response;
+            BulkItemRequest replicaRequest = request.items()[requestIndex];
```

Comment:
this makes sense, made `replicaRequest` final and explicitly set it for all the cases